controllers/front/callbacknotification.php

```diff
@@ -238,4 +253,62 @@ public function postProcess()
         unlockCallback($lockFileName, $lockFileHandle);
         exit($message);
     }
+
+    public function processChildOrder($cart, $response, $transactionStatus, $auth_statuses, $captured_statuses, $order_state, $lockFileName, $lockFileHandle)
+    {
+        $shopOrderId = $response->shopOrderId;
+        // Check if an order exist
+        $order = getChildOrderFromUniqueId($shopOrderId);
+        if (!Validate::isLoadedObject($order)) {
+            $order_id = Order::getOrderByCartId((int) ($cart->id));
+            $orderDetail = new Order((int) $order_id);
```

Comment:
Casting is not enough, because you might cast to 0 for a non-numeric value, and have a side effect here.

You need to ensure that the value is a numeric value, and otherwise fail

Reply:
I found out that these lines were unnecessary so I've removed them.